src/data/auth.ts

```diff
@@ -22,6 +23,20 @@ export interface SignedPath {
 
 export const hassUrl = `${location.protocol}//${location.host}`;
 
+export const autofillAuthSchema = (schemas: HaFormSchema[]) =>
```

Comment:
We should not change these schema's in the frontend, but instead just update them in the backend

Reply:
Yeah I wasn't sure what the preferred approach would be.  Since this is purely for an HTML attribute I thought maybe keeping in the frontend would be better.

Reply:
@bramkragten I finally figured out that extra meta data can be shoved into the `description` field added to voluptuous by @balloob years ago.  After combing through the core though, I'd suggest it'd be best to keep this here.

My logic is that we should also be adding this feature when integrations collect usernames and passwords in the config flow.  By my search, there are 237 which collect "username" and 278 collecting "password".  So if it has to be in the schema from core, all those integrations and each auth provider need to be updated (plus new ones), versus just a couple places to transform it in the frontend.

The frontend essentially already does the same thing by deciding whether to mask a field based on its name (i.e. `type="password"`), instead of that having to be in the schema from core.

Reply:
We should move to selectors, then everything is supported

Reply:
So are you saying all of those flows need to be updated to selectors in order to get this feature?  The [developer docs for data entry flows](https://developers.home-assistant.io/docs/data_entry_flow_index/#show-form) make it seem like neither is preferred.

Reply:
Why does it seem like that? Selectors have more options than the voluptuous schema used now. Since you added this feature to selectors, it makes sense to update those to selectors in the backend too.

Reply:
Because the docs say nothing of moving from voluptuous to only selectors

> You define the current step, the schema of the data (using voluptuous or selectors) and optionally a dictionary of errors.

Then the first example just uses voluptuous not selectors to get a username and password.

Updating the authentication providers is one thing, but it seems excessive to wait for 250+ integrations to convert to selectors in order to support browser autofill.  I would propose:

1. For voluptuous, infer the `input` `type` and `autocomplete` values from the `name` as is already done.  I can add documentation somewhere on this.
2. For text selectors, integration authors need to provide the `autofill` property just as they do the `type`.  I can add to the [text selector documentation](https://www.home-assistant.io/docs/blueprint/selectors/#text-selector).

Reply:
Wow, how come we are suddenly talking about 250+ integrations? I thought this was just about the MFA and login providers?

Reply:
I pushed a change to demonstrate.  Now when an integration asks for username & password credentials in their setup, `autocomplete` will be automatically added there too, which is where the 250+ other schemas come in.